bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -1068,16 +1054,10 @@ void errorOutPendingOps(int rc) {
      */
 
     void errorOutOutstandingEntries(int rc) {
-        // DO NOT rewrite these using Map.Entry iterations. We want to iterate
-        // on keys and see if we are successfully able to remove the key from
-        // the map. Because the add and the read methods also do the same thing
-        // in case they get a write failure on the socket. The one who
-        // successfully removes the key from the map is the one responsible for
-        // calling the application callback.
-        for (CompletionKey key : completionObjectsV2Conflicts.keySet()) {
-            while (completionObjectsV2Conflicts.get(key).size() > 0) {
-                errorOut(key, rc);
-            }
+        Optional<CompletionKey> multikey = completionObjectsV2Conflicts.getAnyKey();
```

Comment:
this 'getAnyKey()' sounds a bit weird to me, therefore inside the implementation we allocating a bunch of objects (using stream API).
I have an alternative proposal:

`completionObjectsV2Conflicts.forEachValue( k -> errorOut(k, rc))`

the iteration will be internal to the implementation, we will save method calls and allocations and overall the code will look better

Reply:
errorOut modifies the collection, so forEachValue will give you a ConcurrentModificationException. 

Regarding allocations, this isn't in a critical path, so I didn't take any care to avoid them, but escape analysis should get rid of most of them in any case.

Reply:
Oh, I missed the internals of errorOut.

Okay for me